### PR TITLE
adds new time diff function

### PR DESCRIPTION
**Pull request by matt-raffel-kiva**
Signed-off-by: Matt Raffel <mattr@kiva.org>

| 🔥 | 🐞 | 🙋 | 🚫 | 🚀 |
|----|----|----|----|----|
|       |       |      |       |       |

*([Click here](https://github.com/kiva/protocol/blob/master/PULL_REQUEST_README.md) if you don't understand these emojis)*

**What issue is this targeting?**
PRO-2325

**Changes proposed in this pull request**
We use an anonymous method to compute delta in two Date types in a lot of places.  Moving it protocol common

**🚀 Deployment changes 🚀**  
(_list added or removed env, db changes etc_)  

**Other Notes**

### CODE REVIEW

File: src/protocol.utility.ts [hunk 1]
```diff
@@ -10,4 +10,23 @@ export class ProtocolUtility {
...
+        eg:
+        ```
+                const startOf = new Date();
+                const deltaMS = timeDelta(new Date(), startOf));
+        ```
+     */
+    public static timeDelta(newest: Date, oldest: Date): number {
+        let result = newest.getTime() - oldest.getTime();
+        if (result <= 0) {
+            result = (result + 1000) % 1000;
```
Comment (voutasaurus) [inline src/protocol.utility.ts:27]: If result <=0 then the caller has put the values in the wrong order, right? Feels like an error, or maybe we want to return the absolute value. Not sure what the `result = (result + 1000) % 1000; ` is supposed to achieve?

Comment (matt-raffel-kiva) [reply src/protocol.utility.ts:27]: I dont really remember.  This logic is copied from the multiple places we use in it protocol and else where. https://github.com/kiva/protocol/blob/1156688e22bd185dd9d56e6cfdad1dc5d4aaa748/wallet_sync_service/src/sync/base-sync.service.ts#L185.  At this time, I would rather leave the logic in place so that we have parity with the code we are making common

Comment (matt-raffel-kiva) [reply src/protocol.utility.ts:27]: I counted at least 5 places this logic is used just in protocol.  It is also used in our aries projects.  It needs to be a common method.

Comment (voutasaurus) [reply src/protocol.utility.ts:27]: Thanks Matt, I’m movable on this. It seems weird though so let’s create another ticket to review / document / understand it and maybe make a unit test that proves the edge case that’s being catered to

**Review by voutasaurus — APPROVED** [review]
lgtm
